store/mongo/datastore_mongo.go

```diff
@@ -325,6 +328,20 @@ var (
 		Options: mopts.Index().
 			SetName(IndexDeviceDeploymentsLogs),
 	}
+
+	// 1.2.13
+	IndexArtifactProvidesRootfsVersionAndChecksum = mongo.IndexModel{
+		Keys: bson.D{
+			{Key: model.StorageKeyImageProvidesRootFSVersion,
+				Value: 1},
+			{Key: model.StorageKeyImageProvidesRootFSChecksum,
+				Value: 1},
+		},
+		Options: &mopts.IndexOptions{
+			Background: &_false,
+			Name:       &IndexArtifactProvidesRootfsVersionAndChecksumName,
```

Comment:
This index needs to be sparse or use a `PartialFilterExpression` to cover documents without Provides.

Reply:
it's a regular index (not unique), so, as you pointed out, there is no need for PartialFilterExpression